modules/reitit-ring/src/reitit/ring.cljc

```diff
@@ -133,21 +133,38 @@
                       " Use :reitit.ring/default-options-endpoint instead.")))
      (r/router data opts))))
 
+(defn- comp-handlers
+  "Compose two ring handlers such that if the first has an empty response
+  the second will be invoked."
+  ([handler]
+   handler)
+  ([handler1 handler2]
+   (let [single-arity (fn [request]
+                        (or (handler1 request) (handler2 request)))
+         multi-arity (fn [request respond raise]
+                       (handler1 request (fn [response]
+                                           (if response
+                                             (respond response)
+                                             (handler2 request respond raise))) raise))]
+     (fn
+       ([request]
+        (single-arity request))
+       ([request respond raise]
+        (multi-arity request respond raise))))))
+
 (defn routes
```

Comment:
I think this could be simplified a bit.

Reply:
In what way?

Reply:
`comp-handlers` has 2 arities, `routes` 4. `routes` is not needed at request-time (thanks for the fix!) so the creation speed is not critical. dropped extra arities here: https://github.com/metosin/reitit/pull/696